src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py

```diff
@@ -1138,3 +1138,283 @@ def list_container_services(client, resource_group_name=None):
+            public_ip_address_id)
+        nic_resource['dependsOn'] = nic_dependencies
+        master_template.add_resource(nic_resource)
+    else:
+        # Using an existing NIC
+        invalid_parameters = [nsg, public_ip_address, subnet, vnet_name]
+        if any(invalid_parameters):
+            raise CLIError('When specifying an existing NIC, do not specify NSG, '
+                           'public IP, VNet or subnet.')
+
+    if storage_profile in [StorageProfile.SACustomImage, StorageProfile.SAPirImage]:
+        os_vhd_uri = 'https://{}.blob.{}/{}/{}.vhd'.format(
+            storage_account, CLOUD.suffixes.storage_endpoint, storage_container_name, os_disk_name)
+
+    vm_resource = build_vm_resource(
+        vm_name, location, tags, size, storage_profile, nics, admin_username, availability_set,
+        admin_password, ssh_key_value, ssh_dest_key_path, image, os_disk_name,
+        os_type, storage_caching, os_publisher, os_offer, os_sku, os_version,
+        os_vhd_uri)
+    vm_resource['dependsOn'] = vm_dependencies
+
+    master_template.add_resource(vm_resource)
+
+    template = master_template.build()
+
+    # deploy ARM template
+    deployment_name = 'vm_deploy_' + random_string(32)
+    client = get_mgmt_service_client(ResourceManagementClient).deployments
+    properties = DeploymentProperties(template=template, parameters={}, mode='incremental')
+    if validate:
+        return client.validate(resource_group_name, deployment_name, properties, raw=no_wait)
+
+    # creates the VM deployment
+    if no_wait:
+        return client.create_or_update(
+            resource_group_name, deployment_name, properties, raw=no_wait)
+    else:
+        LongRunningOperation()(client.create_or_update(
+            resource_group_name, deployment_name, properties, raw=no_wait))
+
+    output_template = ArmTemplateBuilder()
+    output_template.add_output('id', vm_name, 'Microsoft.Compute', 'virtualMachines',
+                               output_type='string', path='vmId')
+    if nic_name:
+        output_template.add_output('privateIpAddress', nic_name, 'Microsoft.Network',
+                                   'networkInterfaces', output_type='string',
+                                   path='ipConfigurations[0].properties.privateIPAddress')
+        output_template.add_output('macAddress', nic_name, 'Microsoft.Network',
+                                   'networkInterfaces', 'string', 'macAddress')
+    if public_ip_address:
+        output_template.add_output('publicIpAddress', public_ip_address, 'Microsoft.Network',
+                                   'publicIPAddresses', output_type='string', path='ipAddress')
+    if public_ip_address_dns_name:
+        output_template.add_output('fqdn', public_ip_address, 'Microsoft.Network',
+                                   'publicIpAddresses', 'string', 'dnsSettings.fqdn')
+    # TODO: Re-enable resource group output
```

Comment:
No, I'm putting it back in. I would like you to look at that change once it's done.